supabase/functions/download_link/index.ts

```diff
@@ -8,35 +8,50 @@ interface DataDownload {
   storage_provider: string
   user_id?: string
   bucket_id: string
+  api_key?: string
 }
 
 serve(async (event: Request) => {
   if (event.method === 'OPTIONS')
     return sendOptionsRes()
+
   const authorization = event.headers.get('authorization')
-  if (!authorization)
-    return sendRes({ status: 'Cannot find authorization' }, 400)
+  if (!authorization) {
+    return sendRes({ status: 'Cannot find authorization token in header' }, 400)
+  }
+    
   // TODO: fix for admin
   try {
     const { data: auth, error } = await supabaseAdmin().auth.getUser(
       authorization?.split('Bearer ')[1],
     )
-    // console.log('auth', auth)
-    if (error || !auth || !auth.user)
-      return sendRes({ status: 'not authorize' }, 400)
 
+    // console.log('auth', auth)
+    let cliUserId = undefined
     const body = (await event.json()) as DataDownload
-    const admin = await isAdmin(auth.user.id)
-    const userId = (admin && body.user_id) ? body.user_id : auth.user.id
+    if (error || !auth || !auth.user) {
+      if (body.api_key) {
+        cliUserId = await verifyUser(body.api_key)
+        if (cliUserId) {
+          console.assert(body.user_id == cliUserId)
+        } else {
+          return sendRes({ status: 'Not authorized. The API key is invalid' }, 401)
```

Comment:
This will never fail, you allays return from verifyUser

Reply:
Right, but if the [`userId` is empty](https://github.com/ayewo/capgo/blob/feature/partial-update/supabase/functions/_utils/supabase.ts#L607-L612), then this block can fail.